conversations/impl/src/main/java/org/sakaiproject/conversations/impl/ConversationsServiceImpl.java

```diff
@@ -477,24 +477,26 @@ public TopicTransferBean saveTopic(final TopicTransferBean topicBean) throws Con
             ConversationsEvents event = isNew ? ConversationsEvents.TOPIC_CREATED : ConversationsEvents.TOPIC_UPDATED;
             eventTrackingService.post(eventTrackingService.newEvent(event.label, decoratedBean.reference, decoratedBean.siteId, true, NotificationService.NOTI_OPTIONAL));
 
-            try {
-                Site site = siteService.getSite(decoratedBean.siteId);
-                Set<User> users = new HashSet<>(userDirectoryService.getUsers(site.getUsers()));
-
-                Map<String, Object> replacements = new HashMap<>();
-                replacements.put("siteTitle", site.getTitle());
-                replacements.put("topicTitle", decoratedBean.title);
-                replacements.put("topicUrl", decoratedBean.portalUrl);
-                replacements.put("bundle", new ResourceLoader("conversations_notifications"));
-
-                userMessagingService.message(users,
-                    Message.builder()
-                        .siteId(decoratedBean.siteId)
-                        .tool(TOOL_ID)
-                        .type(topic.getType() == TopicType.QUESTION ? "newquestion" : "newdiscussion").build(),
-                    Arrays.asList(new MessageMedium[] {MessageMedium.EMAIL}), replacements, NotificationService.NOTI_OPTIONAL);
-            } catch (IdUnusedException iue) {
-                log.error("No group for site reference {}", siteRef);
+            if (sendMessage) {
+                try {
+                    Site site = siteService.getSite(decoratedBean.siteId);
+                    Set<User> users = new HashSet<>(userDirectoryService.getUsers(site.getUsers()));
+
+                    Map<String, Object> replacements = new HashMap<>();
+                    replacements.put("siteTitle", site.getTitle());
+                    replacements.put("topicTitle", decoratedBean.title);
+                    replacements.put("topicUrl", decoratedBean.portalUrl);
+                    replacements.put("bundle", new ResourceLoader("conversations_notifications"));
+
+                    userMessagingService.message(users,
+                        Message.builder()
+                            .siteId(decoratedBean.siteId)
+                            .tool(TOOL_ID)
+                            .type(topic.getType() == TopicType.QUESTION ? "newquestion" : "newdiscussion").build(),
+                        Arrays.asList(new MessageMedium[] {MessageMedium.EMAIL}), replacements, NotificationService.NOTI_OPTIONAL);
```

Comment:
Would it make sense to pass NotificationService.NOTI_NONE or NOTI_IGNORE instead of adding the 'if'?

Reply:
Basically, all this comes from the site seed job running on the nightlies and firing off thousands of emails. I've added some topics and posts to the seed job, and that's the problem :)

Reply:
Bernie that's good point may make more sense to add an api method to control notifications during certain times like running a job creating test data.